### ISSUE TITLE
Discussion: documentation

### ISSUE THREAD

**Issue by ericblade** [ericblade/mws-advanced#27]
Although I originally set out to try really hard to keep the documentation up to date, and useful, inspiration and coding outstripped the pace of documentation updating.  Also, I'm really not at all happy with the documentation that I've found in almost every project out there, so keeping up on the documentation has absolutely been a chore, rather than enjoyable.  

That said, I discovered a project that I think has really great documentation: http://docs.sequelizejs.com/

I'm switching to using sequelize for my main project, that incorporates mws-advanced, and other than some nitpicks as to not being able to quickly locate certain pieces in it, Sequelize's documentation seems to be quite a bit better than most everything else out there.  It really has a whole lot of what I want in a documentation for anything -- quick information for those with the ability to dive right in, great examples, pretty decent in depth info, and it's really easy to read through.  In fact, I *enjoyed* reading the documentation for it.  My only complaint about it is that there's a little bit less of the "what is this useful for?" in some places, and I tried to search for some ability that seemed completely obvious to me, but took some search-fu to figure out how to do it.  (namely, how to create a database index on a model, that used the relationships to other models)

I'm looking into how they build their documentation right now.  Looks like they are using something to build html from .md files.

**Comment by ericblade** [comment]
Looks like they use a thing called "esdoc". I'm looking into it now.

**Comment by ericblade** [comment]
"esdoc" has been implemented.  For the most part, it's way better at it's task than "documentationJS" seemed to be.  I've attached the new documentation (WORK IN PROGRESS) to http://ericblade.github.io/mws-advanced . 
I'm keeping the DocumentationJS generated doc around for a little bit, because esdoc doesn't output to md format, but I'll probably delete it pretty quickly, once the html documentation is improved.
